cmd/swarmctl/service/flagparser/container.go

```diff
@@ -14,6 +14,14 @@ func parseContainer(flags *pflag.FlagSet, spec *api.ServiceSpec) error {
 		spec.Task.GetContainer().Image = image
 	}
 
+	if flags.Changed("hostname") {
+		hostname, err := flags.GetString("hostname")
+		if err != nil {
+			return err
+		}
+		spec.Task.GetContainer().Hostname = hostname
```

Comment:
@vdemeester  Just a small question for my knowledge . 
Why not adding a setter in ```ContainerSpec``` ?